### PR TITLE
Fuzz tokenizer.explain: draft for fuzzy tests.

### PR DESCRIPTION
**Pull request by rmitsch**
## Goals

Fuzz tests for `Tokenizer.explain()` to catch divergences between tokenizer explanations and the actual tokenization results.

## Description

Includes a new test for `Tokenizer.explain()`, generating test sentences with interpunctation fuzzily utilizing `hypothesis`.

### Types of change

New test.

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [ ] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

**Comment by adrianeboyd** [comment]
I think that all the languages in `LANGUAGES_WITH_CUSTOM_TOKENIZERS` are already excluded in the list, so it'd be fine to just use `LANGUAGES` directly. I would xfail this test for now, and a second PR can address the explain bugs so it passes.

**Comment by rmitsch** [comment]
> I think that all the languages in `LANGUAGES_WITH_CUSTOM_TOKENIZERS` are already excluded in the list, so it'd be fine to just use `LANGUAGES` directly. I would xfail this test for now, and a second PR can address the explain bugs so it passes.

Done. Issues for fixing explainer bugs has been added.

**Comment by rmitsch** [comment]
Fetching  the tokenizer (tried `spacy.blank(lang).tokenizer` and `get_lang_class(lang)().tokenizer`) / calling `tokenizer.explain()` [fails in the Azure runs](https://dev.azure.com/explosion-ai/Public/_build/results?buildId=17567&view=logs&j=8752acd4-67f3-58f7-18d5-420e8425f3a4&t=4714c79d-8d5d-55af-5921-ca1da54c42cb&l=14). Seems odd to me, can't reproduce it locally. Any ideas why that may be the case?

File: spacy/tests/tokenizer/test_explain.py [hunk 1]
```diff
@@ -77,3 +83,46 @@ def test_tokenizer_explain_special_matcher(en_vocab):
...
+@pytest.mark.parametrize("lang", LANGUAGES)
+@hypothesis.given(sentence=sentence_strategy())
+def test_tokenizer_explain_fuzzy(lang: str, sentence: str) -> None:
+    """
+    Tests whether output of tokenizer.explain() matches tokenizer output. Input generated by hypothesis.
+    lang (str): Language to test.
+    text (str): Fuzzily generated sentence to tokenize.
+    """
+
+    tokenizer = spacy.blank(lang).tokenizer
```
Comment (adrianeboyd) [inline spacy/tests/tokenizer/test_explain.py:125]:
```suggestion
    tokenizer: Tokenizer = spacy.blank(lang).tokenizer
```

**Pull request merged**